### DISCUSSION TITLE
GET and PUT inconsistency

### DISCUSSION THREAD

**Discussion by cjonasw** [octokit/octokit.js#1724] [General] [answered]
Hi, not sure if there is something I am missing here but I'm attempting to do GET branch/protection and then modify the response to then make a PUT to the same endpoint.

When I do `GET /repos/:owner/:repo/branches/:branch/protection`, the response I get is an inconsistent data structure to that of the request I am expected to make for the equivalent `PUT` (`PUT /repos/:owner/:repo/branches/:branch/protection`).

Example: 

`allow_deletions: { enabled: false }`

and then in the request I have to do this:

`allow_deletions: true`

This is simple for the boolean types but isn't maintainable for the objects and the PUT requires atleast the following due to the error message I get when attempting to supply a partial request: 

> `"enforce_admins", "required_pull_request_reviews", "required_status_checks", "restrictions" weren't supplied.`

There must be something I'm missing here, a mapping method or am I simply approaching this the wrong way?

Any help appreciated.

**Comment by gr2m** [comment, marked as answer]
I'm afraid there is not much we can do with the SDK to help. We just wrap the existing REST API endpoints, we are not changing them, with a very few exceptions. You can see the inconsistencies documented in the REST API here: https://developer.github.com/v3/repos/branches/#update-branch-protection. I'm afraid you will have to handle the inconsistencies yourself.

One alternative would be to use GraphQL. I quickly checked, but it looks to me like the `allowDeletions` settings is not enabled yet. You could check in with https://support.github.com/contact to ask about that.

Yet another alternative would be to create an Octokit plugin which abstracts away these differences. I'm sure you are not the first to run into that problem, creating a reusable plugin might be useful for other people facing the same problem. Also if your project is private, you could make the plugin code public, which would allow me and others to collaborate on your code. Creating a reusable plugin is fairly simple: https://octokit.github.io/rest.js/v17#plugins. And I created an `npm init` script to bootstrap a plugin repository: https://github.com/octokit/create-octokit-project.js/

**Reply by cjonasw** [reply]
Thanks for the response. That's a shame, thanks for advice though.

Ah okay, so the allow deletions setting within this endpoint is not yet available anyway?

**Reply by gr2m** [reply]
the allow deletions setting is enabled for the REST API endpoints, but I don't see it in the GraphQL schema.
